doc/source/ray-core/scheduling/resources.rst

```diff
@@ -0,0 +1,166 @@
+while the system manages scheduling and autoscaling based on resource requests.
+
+A resource in Ray is a key-value pair where the key denotes a resource name, and the value is a float quantity.
+For convenience, Ray has native support for CPU, GPU, and memory resource types; CPU, GPU and memory are called **pre-defined resources**.
+Besides those, Ray also supports :ref:`custom resources <custom-resources>`.
+
+Physical Resources and Logical Resources
+----------------------------------------
+
+Physical resources are resources that a machine physically has such as physical CPUs and GPUs
+and logical resources are virtual resources defined by a system.
+
+Ray resources are **logical** and don’t need to have 1-to-1 mapping with physical resources.
+For example, you can start a Ray head node with 3 GPUs via ``ray start --head --num-gpus=3`` even if it physically has zero.
+They are mainly used for admission control during scheduling.
+
+The resources being logical has several implications:
+
+- Resource requirements of tasks or actors does NOT impose any limits on the actual physical resource usage.
+  For example, Ray doesn't prevent a ``num_cpus=1`` task from launching multiple threads and using multiple physical CPUs.
+  It's your responsibility to make sure your tasks or actors use no more resources than specified via resource requirements.
+- Ray doesn't provide CPU isolation for tasks or actors.
+  For example, Ray won't reserve a physical CPU excludsively and pin a ``num_cpus=1`` task to it.
+  Ray will let the OS scheduler schedule and run the task instead.
+  If needed, you can use OS APIs like ``sched_setaffinity`` to pin a task to a physical CPU.
+- Ray does provide :ref:`GPU <gpu-support>` isolation in forms of visible devices by automatically setting the ``CUDA_VISIBLE_DEVICES`` envrionment variable,
+  but it is still up to the task or actor to actually honor it or override it.
+
+.. figure:: ../images/physical_resources_vs_logical_resources.svg
+
+  Physcial resources vs logical resources
+
+.. _custom-resources:
+
+Custom Resources
+----------------
+
+Besides pre-defined resources, you can also specify custom resources that a Ray node has and request them in your tasks or actors.
+Several use cases of custom resources are:
+
+- Your node has special hardware and you can represent it as a custom resource.
+  Then your tasks or actors can request the custom resource via ``@ray.remote(resources={"special_hardware": 1})``
+  and Ray will schedule the tasks or actors to the node that has the custom resource.
+- You can use custom resources as labels to tag nodes and you can achieve label based affinity scheduling.
+  For example, you can do ``ray.remote(resources={"custom_label": 0.001})`` to schedule tasks or actors to nodes with ``custom_label`` custom resource.
+  In this use case, the actual quantity doesn't matter that much and the convention is to specify a tiny number so that the label resource is
```

Comment:
```suggestion
  For this use case, the actual quantity doesn't matter, and the convention is to specify a tiny number so that the label resource is
```